utils_cv/detection/model.py

```diff
@@ -167,109 +185,112 @@ def fit(self, epochs: int, print_freq: int = 10) -> None:
-            idx: the index of the dataset to visualize
-            ax: axes to draw on
+            dl: the dataloader to predict on
+            threshold: iou threshold for a positive detection
 
-        Returns nothing but plots graph.
+        Returns a list of AnnotationBboxes
         """
-        im_path = (
-            self.dataset.root
-            / self.dataset.image_folder
-            / self.dataset.ims[idx]
-        )
-        im = Image.open(str(im_path))
-
-        # plot prediction boxes
-        pred = self.inference(im)
-        pred_labels, pred_boxes = get_bounding_boxes(pred)
-        pred_params = PlotSettings(rect_color=(255, 0, 0), text_size=0)
-        im = plot_boxes(
-            im,
-            pred_boxes,
-            [self.dataset.categories[l] for l in pred_labels],
-            plot_settings=pred_params,
-        )
+        pred_generator = self.predict_batch(dl, threshold=threshold)
+        detections = [pred for preds in pred_generator for pred in preds]
+        return detections
 
-        # plot ground truth boxes
-        ground_truth_params = PlotSettings(rect_color=(0, 255, 0), text_size=0)
-        boxes, categories, im_path = self.dataset.get_image_features(idx)
-        im = plot_boxes(
-            im, boxes, categories, plot_settings=ground_truth_params
-        )
+    def predict_batch(
+        self, dl: DataLoader, threshold: int = 0.6
+    ) -> Generator[List[AnnotationBbox], None, None]:
+        """ Batch predict
 
-        # show image
-        ax.set_xticks([])
-        ax.set_yticks([])
-        ax.imshow(im)
+        Args
+            dl: A DataLoader to load batches of images from
+            threshold: iou threshold for a positive detection
 
-    def show_preds(self, ds: Dataset = None, rows: int = 1) -> None:
-        """ Show batch of predictions against ground truth.
+        Returns an iterator that yields a list of annotations for each image that is scored
+        """
 
-        Args:
-            ds: the datset to use, by default, use test_ds
-            rows: rows to predict
+        labels = self.dataset.labels
+        model = self.model.eval()
 
-        Returns nothing
-        """
-        if ds is None:
-            ds = self.test_ds
+        for i, batch in enumerate(dl):
+            ims, infos = batch
+            ims = [im.cuda() for im in ims]
+            with torch.no_grad():
+                preds = model(list(ims))
 
-        # setup iterator if not yet setup
-        if not hasattr(self, "ds_iterator"):
-            self.ds_iterator = iter(ds.indices)
+            bbox_batch = []
+            for pred, info in zip(preds, infos):
 
-        plot_grid(
-            self.show_detection_vs_ground_truth,
-            partial(next, self.ds_iterator),
-            rows=rows,
-        )
+                im_idx = int(info["image_id"].numpy())
+                im_path = dl.dataset.dataset.get_path_from_idx(im_idx)
+
+                anno_bboxes = get_bounding_boxes(
```

Comment:
Technically this is not an annotation bbox, but a prediction bbox. Maybe just rename to bboxes since it's clear in this context?